src/main/java/org/broadinstitute/hellbender/tools/walkers/ReferenceConfidenceVariantContextMerger.java

```diff
@@ -379,19 +388,42 @@ private void addReferenceConfidenceAttributes(final VCWithNewAlleles vcPair, fin
                     annotationMap.put(key, values);
                 }
                 try {
-                    values.add(parseNumber(value.toString()));
+                    //TODO: Verify that the code paths that lead here guarantee that these are always INFO attributes ?
+                    values.add(parseNumericInfoAttributeValue(vcfInputHeader, key, value.toString()));
                 } catch (final NumberFormatException e) {
                     warning.warn(String.format("Detected invalid annotations: When trying to merge variant contexts at location %s:%d the annotation %s was not a numerical value and was ignored",vcPair.getVc().getContig(),vcPair.getVc().getStart(),p.toString()));
                 }
             }
         }
     }
 
-    private Comparable<?> parseNumber(String stringValue) {
-        if (stringValue.contains(".")) {
-            return Double.parseDouble(stringValue);
-        } else {
-            return Integer.parseInt(stringValue);
+    // Use the VCF header's declared type for the given attribute to ensure that all the values for that attribute
+    // across all the VCs being merged have the same boxed representation. Some VCs have a serialized value of "0"
+    // for FLOAT attributes, with no embedded decimal point, but we still need to box those into Doubles, or the
+    // subsequent sorting required to obtain the median will fail due to the list having a mix of Comparable<Integer>
+    // and Comparable<Double>.
+    private Comparable<?> parseNumericInfoAttributeValue(final VCFHeader vcfHeader, final String key, final String stringValue) {
+        final VCFInfoHeaderLine infoLine = vcfHeader.getInfoHeaderLine(key);
+        if (infoLine == null) {
+            warning.warn(String.format("No header line for INFO attribute %s was found in the input header", key));
```

Comment:
This warning seems problematic. If it gets triggered it will probably get spit out for EVERY read in a VCF which would result in a torrential span. Perhaps us a OneShotLogger here to save peoples poor poor bash histories...

Reply:
Looking at the code leading up to this I don't expect this warning will be a problem for any non-reducible annotations except i'm not sure i completely trust our VCFStandardHeaderLines to have the correct types. We have a number of issues with the header lines either here or in HTSJDK being wrong. This is still better I would say.

Reply:
Excellent point - but `warning` *is* a `OneShotLogger`. Also, I did keep the fallback code for the old behavior if the header line isn't found.

Reply:
Right, good point about `warning`. The next point is that currently its just a dumb implementation that only emits once. If there are multiple error states in a file its probably best to open multiple loggers so if both are triggered they will each be written out, warning is currently used for a different case to do with merging and numerical parsing. maybe split them into 2 loggers?

Reply:
Right. Split out into a second logger, with a slightly more generic message.

Reply:
Also, I rebased this on top of my master to pick up my last CombineGVCFs branch/fix.